src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -248,7 +248,9 @@ public Task Task
                     }
                 }
 
-                return this.wrappedTask as Task ?? this.GetTaskFromCompletionSource(this.wrappedTask);
+                // Read 'wrappedTask' once, so we know it's value is either a Task or a TaskCompletionSource.
+                var wrappedTask = this.wrappedTask;
+                return wrappedTask as Task ?? this.GetTaskFromCompletionSource(wrappedTask);
```

Comment:
📝 Testing is possible by overriding `GetTaskFromCompletionSource`, but this project does not allow IVTs for testing.